rocAL/include/loaders/audio/audio_loader.h

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright (c) 2019 - 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#pragma once
+
+#include <string>
+#include <thread>
+#include <vector>
+#include "commons.h"
+#include "circular_buffer.h"
+#include "audio_read_and_decode.h"
+#include "meta_data_reader.h"
+
+// AudioLoader runs an internal thread for loading an decoding of audios asynchronously
+// It uses a circular buffer to store decoded audios for the user
+class AudioLoader : public LoaderModule {
+public:
+    explicit AudioLoader(void* dev_resources);
+    ~AudioLoader() override;
+    LoaderModuleStatus load_next() override;
+    void initialize(ReaderConfig reader_cfg, DecoderConfig decoder_cfg, RocalMemType mem_type, unsigned batch_size, bool keep_orig_size = false) override;
```

Comment:
What is the use of keep_orig_size

Reply:
We are not making use of this variable, its a part of the function implementation since the Base class has it defined as a pure virtual function .
Please note @fiona-gladwin .